### PR TITLE
#203 Fixing the support for GZIP and DEFLATE saved payloads.

### PR DESCRIPTION
**Pull request by cowboygneox**

### CODE REVIEW

**Comment by cowboygneox** [comment]
Resolves #203.

File: betamax-tests/src/test/groovy/software/betamax/tape/ContentEncodingSpec.groovy [hunk 1]
```diff
@@ -47,7 +49,13 @@ class ContentEncodingSpec extends Specification {
...
         response.addHeader(CONTENT_TYPE, PLAIN_TEXT_UTF_8.withoutParameters().toString())
         response.addHeader(CONTENT_ENCODING, encoding)
-        response.body = "O HAI!".bytes
+        if (encoding == "gzip") {
+            response.body = new GzipEncoder().encode("O HAI!")
+        } else if (encoding == "deflate") {
+            response.body = new DeflateEncoder().encode("O HAI!")
+        } else {
+            response.body = "O HAI!".bytes
+        }
```
Comment (faucct) [inline betamax-tests/src/test/groovy/software/betamax/tape/ContentEncodingSpec.groovy]: Does not switch look better here?

File: betamax-core/src/main/java/software/betamax/tape/MemoryTape.java [hunk 2]
```diff
@@ -272,7 +274,14 @@ private void recordResponseBody(Response response, RecordedResponse recording) t
...
         if (representAsText) {
-            recording.setBody(message.getBodyAsText());
+            // TODO find a better way to accommodate for this behavior
+            if (message.getEncoding().equals("gzip")) {
+                recording.setBody(new GzipEncoder().decode(new ByteArrayInputStream(message.getBodyAsBinary())));
+            } else if (message.getEncoding().equals("deflate")) {
+                recording.setBody(new DeflateEncoder().decode(new ByteArrayInputStream(message.getBodyAsBinary())));
+            } else {
+                recording.setBody(message.getBodyAsText());
+            }
```
Comment (faucct) [inline betamax-core/src/main/java/software/betamax/tape/MemoryTape.java]: `switch` is definitely better here and maybe you should have an enum for a encoding.

**Comment by faucct** [comment]
Seems fine for me. Thanks!

File: betamax-tests/src/test/groovy/software/betamax/tape/ContentEncodingSpec.groovy (hunk 1 above)
Comment (cowboygneox) [inline betamax-tests/src/test/groovy/software/betamax/tape/ContentEncodingSpec.groovy]: @faucct I agree a switch looks better. I was following prior art in the code until I have the time to do a full refactor. This was simply something to drop into place to get a SNAPSHOT out for you to use.

**Comment by cowboygneox** [comment]
@faucct Thanks for your feedback. Merging.

**Pull request merged**